meshtastic/tests/test_node_runtime_seturl.py

```
@@ -0,0 +1,1695 @@
+
+        # Mock admin context
+        admin_context = MagicMock()
+        admin_context.admin_index_for_write = 0
+
+        # First write attempt fails, second succeeds
+        call_count = [0]
+
+        def write_side_effect(_channel: channel_pb2.Channel, _adminIndex: int) -> None:
+            call_count[0] += 1
+            if call_count[0] == 1:
+                raise RuntimeError("Write failed")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)